builder/dockerfile/dispatchers.go

```diff
@@ -631,3 +640,12 @@ func errExactlyOneArgument(command string) error {
 func errTooManyArguments(command string) error {
 	return fmt.Errorf("Bad input to %s, too many arguments", command)
 }
+
+// getShell is a helper function which gets the right shell for prefixing the
+// shell-form of RUN, ENTRYPOINT and CMD instructions
+func getShell(c *container.Config) []string {
+	if 0 == len(c.Shell) {
+		return defaultShell
+	}
+	return c.Shell
+}
```

Comment:
Since we use the value of this `getShell()` function canonically as the first argument to `append()`, wouldn't it be prudent to have it return a _new_ slice instead of directly returning the existing slices so we don't inadvertently change either our default or the one in the supplied `runConfig`?  (ie, `return defaultShell[:]` and `return c.Shell[:]`)